### PR TITLE
nexulm fixes

### PR DESCRIPTION
**Pull request by fenugrec**
I have rebased some (all ?) of the commits that had been added to @nexulm's fork.

It affects the Makefile, so when the cmake branch is ready/merged some of this will need to be carried over (just some extra board types).

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -1,244 +1,247 @@
...
+	if(!frame)
+	  return;
+	
+	if (USBD_GS_CAN_SendFrame(&hUSB, frame) == USBD_OK) {
+	        queue_push_back(q_frame_pool, frame);
+	} else {
+	        queue_push_front(q_to_host, frame);
+	}
+}
+
```
Comment (jhofstee) [inline src/main.c]: this changes line endings of the whole file. Which is fine by itself, but should be a different patch, since this makes the whole diff unreadable. I have cleanup patches for that.. Perhaps that should go in first?

**Comment by fenugrec** [comment]
merge in 68c514e8fcb0e682e0e06752bd9cf6c4c09d4ea6
